orm/src/test/java/org/hibernate/search/test/embedded/EmbeddedEntityNotIndexedTest.java

```diff
@@ -6,46 +6,29 @@
  */
 package org.hibernate.search.test.embedded;
 
+import static org.junit.Assert.assertNull;
+
 import java.sql.Timestamp;
+
 import javax.persistence.Entity;
 import javax.persistence.GeneratedValue;
 import javax.persistence.Id;
 import javax.persistence.OneToOne;
 
-import org.hibernate.annotations.common.reflection.java.JavaReflectionManager;
 import org.hibernate.search.annotations.Field;
 import org.hibernate.search.annotations.Indexed;
 import org.hibernate.search.annotations.IndexedEmbedded;
-import org.hibernate.search.cfg.spi.SearchConfiguration;
-import org.hibernate.search.engine.impl.ConfigContext;
-import org.hibernate.search.engine.metadata.impl.AnnotationMetadataProvider;
 import org.hibernate.search.engine.metadata.impl.TypeMetadata;
 import org.hibernate.search.testsupport.TestForIssue;
-import org.hibernate.search.testsupport.setup.BuildContextForTest;
-import org.hibernate.search.testsupport.setup.SearchConfigurationForTest;
-import org.junit.Before;
+import org.hibernate.search.testsupport.setup.AbstractAnnotationMetadataTest;
 import org.junit.Test;
 
-import static org.junit.Assert.assertNull;
-
 
 /**
  * @author Hardy Ferentschik
  */
 @TestForIssue(jiraKey = "HSEARCH-1494")
-public class EmbeddedEntityNotIndexedTest {
-
-	private AnnotationMetadataProvider metadataProvider;
-
-	@Before
-	public void setUp() {
-		SearchConfiguration searchConfiguration = new SearchConfigurationForTest();
-		ConfigContext configContext = new ConfigContext(
-				searchConfiguration,
-				new BuildContextForTest( searchConfiguration )
-		);
-		metadataProvider = new AnnotationMetadataProvider( new JavaReflectionManager(), configContext );
-	}
+public class EmbeddedEntityNotIndexedTest extends AbstractAnnotationMetadataTest {
```

Comment:
Yes, the initialization of the AnnotationMetadataProvider is more complicated now as we need a MutableSearchFactoryState so I made sure it was done only once.